glusterd2/commands/volumes/volume-create.go

```diff
@@ -73,6 +73,25 @@ func createVolinfo(req *api.VolCreateReq) (*volume.Volinfo, error) {
 		v.Type = volume.DistReplicate
 	}
 
+	if req.DisperseData != 0 || req.DisperseRedundancy != 0 {
+		v.Type = volume.Disperse
+		v.DistCount = 1
+		if req.DisperseData == 0 {
+			v.DisperseCount = len(req.Bricks)
+		} else {
+			v.DisperseCount = req.DisperseData
+		}
+		if req.DisperseRedundancy == 0 {
+			v.RedundancyCount = int(getRedundancy(uint(v.DisperseCount)))
+		} else {
+			v.RedundancyCount = req.DisperseRedundancy
+		}
+		if 2*v.RedundancyCount >= len(req.Bricks) {
```

Comment:
This is not the correct condition.
One basic condition we are not checking which is "disperse-data == redundancy"  which should not be allowed.
Go to cli/src/cli-cmd-parser.c and exxplore cli_cmd_create_disperse_check function to understand conditions.
Also, only disperse-data without redundancy count is not allowed.